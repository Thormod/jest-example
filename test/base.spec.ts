// const agent = require('superagent-promise')(require('superagent'), Promise);
import agent = require('superagent');
import statusCode = require('http-status-codes');


describe('Given a website for API testing', () => {
    test('When consume POST service with query parameters', () => {
        const query = {
            name: 'John',
            age: '31',
            city: 'New York'
        };

        return agent
        .post('https://httpbin.org/post')
        .send(query)
        .then((response) => {
            expect(response.status).toBe(statusCode.OK);
            expect(response.body.json).toMatchSnapshot();
        });
    });

    test('When consume GET service with query parameters', () => {
        const query = {
            name: 'John',
            age: '31',
            city: 'New York'
        };

        return agent.get('https://httpbin.org/get')
        .query(query)
        .then((response) => {
            expect(response.status).toBe(statusCode.OK);
            expect(response.body.args).toMatchSnapshot();
        });
    });

    test('When consume DELETE service with query parameters', () => {
        const body = {
          name: 'Sebastián',
          age: 23,
          city: 'Medellin'
        };
    
        return agent
        .del('https://httpbin.org/delete')
        .send(body)
        .then((response) => {
            expect(response.status).toBe(statusCode.OK);
            expect(response.body.json).toMatchSnapshot();
        });
    });

    test('Consume PUT Service with query parameters', () => {
        const body = {
            name: 'Sebastián',
            age: 23,
            city: 'Medellin'
        };

        return agent
        .put('https://httpbin.org/put')
        .send(body)
        .then((response) => {
            expect(response.status).toBe(statusCode.OK);
            expect(response.body.json).toMatchSnapshot();
        });
    });
    
    test('Consume PATCH Service with query parameters', () => {
        const body = {
          name: 'Sebastián',
          age: 23,
          city: 'Medellin'
        };
    
        return agent
        .patch('https://httpbin.org/patch')
        .send(body)
        .then((response) => {
            expect(response.status).toBe(statusCode.OK);
            expect(response.body.json).toMatchSnapshot();
        });
    });
});


