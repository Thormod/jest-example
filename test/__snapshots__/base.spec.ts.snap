// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a website for API testing Consume PATCH Service with query parameters 1`] = `
Object {
  "age": 23,
  "city": "Medellin",
  "name": "Sebastián",
}
`;

exports[`Given a website for API testing Consume PUT Service with query parameters 1`] = `
Object {
  "age": 23,
  "city": "Medellin",
  "name": "Sebastián",
}
`;

exports[`Given a website for API testing When consume DELETE service with query parameters 1`] = `
Object {
  "age": 23,
  "city": "Medellin",
  "name": "Sebastián",
}
`;

exports[`Given a website for API testing When consume GET service with query parameters 1`] = `
Object {
  "age": "31",
  "city": "New York",
  "name": "John",
}
`;

exports[`Given a website for API testing When consume POST service with query parameters 1`] = `
Object {
  "age": "31",
  "city": "New York",
  "name": "John",
}
`;
